File: client/src/app/models/order.ts
export interface ShippingAddress {
    fullName: string;
    address1: string;
    address2: string;
    city: string;
    state: string;
    zip: string;
    country: string;
}

export interface OrderItem {
    productID: number;
    name: string;
    pictureURL: string;
    price: number;
    quantity: number;
    discount: number;
}

export interface Order{
    id: number;
    buyerID: string;
    shippingAddress: ShippingAddress;
    orderDate: string;
    orderItems: OrderItem[];
    subtotal: number;
    deliveryFee: number;
    orderStatus: string;
    total: number;
}

File: API/Entities/OrderAggregate/Order.cs
using System.ComponentModel.DataAnnotations;

namespace API.Entities.OrderAggregate
{
    public class Order
    {
        public int ID { get; set; }
        public string BuyerID { get; set; }

        [Required]
        public ShippingAddress ShippingAddress { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;
        public List<OrderItem> OrderItems { get; set; }
        public long Subtotal { get; set; }
        public long DeliveryFee { get; set; }
        public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
        public string PaymentIntentID { get; set; }

        public long GetTotal()
        {
            return Subtotal + (Subtotal / 100 * 15) + DeliveryFee;
        }
    }
}